File: python/cosasDesorganizadas/otrosAlgoritmos/hanoi.py



def hanoi(numDiscos, origen, auxiliar, destino):

	if numDiscos == 1:
		print 'Mover disco de %s a %s' % (origen, destino)
	else:
		hanoi(numDiscos-1, origen, destino, auxiliar)
		print 'Mover disco de %s a %s' % (origen, destino)
		hanoi(numDiscos-1, auxiliar, origen, destino)


if __name__ == "__main__":

	hanoi(3,"Torre 1","Torre 2","Torre 3")

File: java/ProyectosDiversos/Algoritmos/src/main/java/com/jugando/algoritmos/Hanoi.java
package com.jugando.algoritmos;

public class Hanoi {

    public static void main(String[] args) {
        int n = 2;
        Hanoi(n, "Torre 1", "Torre 2", "Torre 3");  //1:origen  2:auxiliar 3:destino
    }

    public static void Hanoi(int n, String origen, String auxiliar, String destino) {
        if (n == 1) {
            System.out.println("mover disco de " + origen + " a " + destino);
        } else {
            Hanoi(n - 1, origen, destino, auxiliar);
            System.out.println("mover disco de " + origen + " a " + destino);
            Hanoi(n - 1, auxiliar, origen, destino);
        }
    }
}
